contracts/6/forwarder/BiconomyForwarder.sol

```diff
@@ -0,0 +1,162 @@
+pragma solidity ^0.6.8;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import "./ERC20ForwardRequest.sol";
+
+contract BiconomyForwarder is ERC20ForwarderRequest{
+    using ECDSA for bytes32;
+
+    string public constant EIP712_DOMAIN_TYPE = "EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)";
+
+    bytes32 public constant REQUEST_TYPEHASH = keccak256(bytes("ERC20ForwardRequest(address from,address to,address token,address feeReceiver,address feeMultiplierManager,uint256 msgValue,uint256 gas,uint256 price,uint256 nonce,bytes32 dataHash)"));
+
+    mapping(bytes32 => bool) public domains;
+
+    mapping(address => uint256) private nonces;
+
+    // solhint-disable-next-line no-empty-blocks
+    receive() external payable {}
+
+    function getNonce(address from)
+    public view
+    returns (uint256) {
+        return nonces[from];
+    }
+
+    function verifyEIP712(
```

Comment:
It means we can verify signatures in go, instead of querying the nonce and doing the signature verification on the client. "Inspired by" GSN, not going to lie lol